src/apiv2.ts

```diff
@@ -0,0 +1,272 @@
+import fetch, { Response, RequestInit } from "node-fetch";
+
+import { FirebaseError } from "./error";
+import * as logger from "./logger";
+import * as responseToError from "./responseToError";
+import * as scopes from "./scopes";
+
+const CLI_VERSION = require("../package.json").version;
+
+const VALID_METHODS = new Set(["GET", "PUT", "POST", "DELETE", "PATCH"]);
+
+type FirebaseRequestOptions = {
+  method?: string;
+  baseURL: string;
+  path: string;
+  headers?: { [key: string]: string };
+  json?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  responseType?: "json";
+  searchParams?: { [key: string]: string | number };
```

Comment:
Prefer `queryParams` or just `query` (even if that doesn't match `fetch` semantics, I think).